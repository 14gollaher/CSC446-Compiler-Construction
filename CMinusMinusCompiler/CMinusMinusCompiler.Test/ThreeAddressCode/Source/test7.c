const g = 33;
int a,b,c,d,e,f;
int main()
{

  a=5;
  b=10;
  d= g + 20;
  c = (g+d) * (a + b);
  return c;
}
