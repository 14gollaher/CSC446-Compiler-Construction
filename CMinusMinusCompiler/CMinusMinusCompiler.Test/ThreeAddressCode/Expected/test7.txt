PROC main
	_BP-2 = 5
	_a = _BP-2
	_BP-4 = 10
	_b = _BP-4
	_BP-6 = 33
	_BP-10 = 20
	_BP-8 = _BP-6 + _BP-10
	_d = _BP-8
	_BP-12 = 33
	_BP-14 = _BP-12 + _d
	_BP-18 = _a + _b
	_BP-16 = _BP-14 * _BP-18
	_c = _BP-16
	_AX = _c
ENDP main

START PROC main

